### PR TITLE
added logroate function & remove package dependency for cron.absent

### PR DESCRIPTION
**Pull request by jsmhatre**
Added logrotate policy. cron.absent code contain dependency of cadf_packages. cadf_packages will not be added if dispatcher or listener not enabled and without dispatcher and listener config cron is useless. So it will not add cron if its not defined in config and cron not required where where application not installed. It's creating error where dispatcher and listener disabled.

### CODE REVIEW

File: cadf/files/cadf_dispatcher_logrotate.conf [hunk 1]
```diff
@@ -0,0 +1,9 @@
+/var/log/cadf/cadf_dispatcher.log {
```
Comment (epcim) [inline cadf/files/cadf_dispatcher_logrotate.conf]: make the path configurable

File: cadf/dispatcher.sls [hunk 2]
```diff
@@ -42,14 +52,5 @@ cron_path:
     - require:
       - pkg: cadf_packages
 
-{%- else %}
-
-distpatcher_cron:
-  cron.absent:
-    - identifier: cadf_dispatcher
-    - user: "{{ config.get('user','root') }}"
-    - require:
```
Comment (epcim) [inline cadf/dispatcher.sls]: I would only remove the require here.
As cront.absent should not fire anny errors if the task is no longer in list?

File: cadf/dispatcher.sls [hunk 3]
```diff
@@ -23,6 +23,16 @@
     - require:
         - file: {{ config.config_dir }}
 
+/etc/logrotate.d/cadf_dispatcher:
```
Comment (epcim) [inline cadf/dispatcher.sls]: Don'd deploy logrotate config this way, use this approach:
https://github.com/salt-formulas/salt-formula-linux/blob/master/linux/meta/logrotate.yml

**Review by epcim — CHANGES_REQUESTED** [review]
Please dont remove cron.absent.
To ship logrotate configs use meta/logrotate.yml in this formula and install logrotate formula on the node.

File: cadf/files/cadf_dispatcher_logrotate.conf (hunk 1 above)
Comment (jsmhatre) [reply cadf/files/cadf_dispatcher_logrotate.conf]: updated details as suggested

File: cadf/dispatcher.sls (hunk 2 above)
Comment (jsmhatre) [reply cadf/dispatcher.sls]: It's fine with us. As after removed condition for cadf_package (require option), it doent throw error for other nodes where dispatcher disabled.

File: cadf/dispatcher.sls (hunk 3 above)
Comment (jsmhatre) [reply cadf/dispatcher.sls]: I have verified, meta option is used for generic purpose module. here we want to use it for specific service and if we go with metadata option, there will be too many changes in other repo as well. I have also found there are specific logrotate policy written for kubernetes, apaches etc. So please consider this code

**Comment by jsmhatre** [comment]
@epcim @msenin94 can you please review my PR

File: cadf/files/logrotate [hunk 4]
```diff
@@ -0,0 +1,12 @@
+{%- from "cadf/map.jinja" import config with context %}
```
Comment (epcim) [inline cadf/files/logrotate]: remove this file completeley

File: cadf/logrotate.sls [hunk 5]
```diff
@@ -0,0 +1,14 @@
+{%- from "cadf/map.jinja" import config with context %}
```
Comment (epcim) [inline cadf/logrotate.sls]: remove

File: cadf/init.sls [hunk 6]
```diff
@@ -4,6 +4,7 @@
 include:
 - cadf.dispatcher
 - cadf.listener
+- cadf.logrotate
```
Comment (epcim) [inline cadf/init.sls]: not this way: Define logrotate like: https://github.com/salt-formulas/salt-formula-linux/blob/master/linux/meta/logrotate.yml

So in ```cadf/files/logrotate```
salt-formula-cadf/cadf/meta/logrotate.yml``` store the configs you have in

**Review by epcim — CHANGES_REQUESTED** [review]

File: cadf/meta/logrotate.yml [hunk 7]
```diff
@@ -0,0 +1,19 @@
+{%- from "cadf/map.jinja" import config with context %}
```
Comment (epcim) [inline cadf/meta/logrotate.yml]: config,dispatcher,listener (use oneliner)

File: cadf/meta/logrotate.yml [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+{%- from "cadf/map.jinja" import dispatcher with context %}
+{%- from "cadf/map.jinja" import listener with context %}
+
+{%- if dispatcher.enabled or listener.enabled  %}
+job:
+  cadf:
+    - files: {{ config.log_dir }}/*.log
```
Comment (epcim) [inline cadf/meta/logrotate.yml]: can we do a change, backward compatible, and use following pillar structure? 
```
config:
  log:
    dir: /xyz
    logrotate:
       rotate: 7
       minsize: 50M
       maxsize: 100M
      ```

with these defaults in map.jinja

Comment (epcim) [reply cadf/meta/logrotate.yml]: I changed my mind, If you define log_dir fist time, then this way
 ```
config:
  log:
    dir: /xyz
```

This is not required for the merge, so keep the for the other day:
```
logrotate:
       rotate: 7
       minsize: 50M
       maxsize: 100M
```

**Review by epcim — CHANGES_REQUESTED** [review]

File: cadf/meta/logrotate.yml (hunk 7 above)
Comment (jsmhatre) [reply cadf/meta/logrotate.yml]: will update same

**Comment by epcim** [comment]
closed in favor of updated PR#9

**Pull request closed without merging**